pkg/cloud/scope/rosamachinepool.go

```diff
@@ -70,6 +74,18 @@ func NewRosaMachinePoolScope(params RosaMachinePoolScopeParams) (*RosaMachinePoo
 		return nil, errors.Wrap(err, "failed to init MachinePool patch helper")
 	}
 
+	controlPlaneScope := &ROSAControlPlaneScope{
```

Comment:
Let's implement the required interface instead on the machine pool scope?

```
var _ cloud.SessionMetadata = &RosaMachinePoolScope{}
```

Reply:
the MachinePool doesn't doesn't have its own identify reference, so implementing the `IdentityRef()` interface func didn't seem right to me.
Happy to do it if you disagree.

Reply:
It's probably a matter of semantics, but in a sense the `ROSAMachinePool` does have an identity reference - it must always use the reference on the `ROSAControlPlane` that it's associated with. Let's implement the interface here so we don't create and throw away a `ROSAControlPlaneScope{}` - we have all the information necessary to do the lookup in the machine pool scope params.